### PR TITLE
RHTAPBUGS-1226 Add cleanup after test run

### PR DESCRIPTION
**Pull request by jkopriva**

### CODE REVIEW

File: default.env [hunk 1]
```diff
@@ -29,4 +29,10 @@ export RED_HAT_DEVELOPER_HUB_URL=''
 
 # Namespace where ArgoCD deploys the RH Developer Hub components.
 # Required: yes
+# Example: rhtap-app
 export APPLICATION_ROOT_NAMESPACE=''
+
+# RHTAP root namespace where RHTAP is installed.
+# Required: no
+# Default: rhtap
+export RHTAP_ROOT_NAMESPACE='rhtap'
```
[the remark this answers is not in the document]
Comment (jkopriva) [reply default.env:38]: this is the namespace, where you install the rhtap(with helm installer). You can change templates/private-values.yaml to install applications in some custom namespace(like the rhtap-app-prod/stage/development) and install RHTAP in different namespace like in this case: 
image

File: src/apis/kubernetes/kube.ts [hunk 2]
```diff
@@ -255,4 +262,37 @@ export class Kubernetes extends Utils {
...
+
+             // Patch the app
+            await k8sCoreApi.patchNamespacedCustomObject('argoproj.io','v1alpha1', namespace, 'applications', applicationName,  patchObject, undefined, undefined, undefined, options);
+
+            // Delete the app
+            await k8sCoreApi.deleteNamespacedCustomObject('argoproj.io','v1alpha1', namespace, 'applications', applicationName)
+
+            console.log(`App ${applicationName} patched and deleted successfully.`);
+        } catch (error) {
+            console.error('Error:', error);
```
Comment (flacatus) [inline src/apis/kubernetes/kube.ts:295]: throw new Error()...

File: tests/gpts/github/test-config/github_advanced_scenario.ts [hunk 3]
```diff
@@ -406,5 +408,12 @@ export const githubSoftwareTemplatesAdvancedScenarios = (gptTemplate: string) =>
                 throw new Error("Component seems was not synced by ArgoCD in 10 minutes");
             }
         }, 900000)
+
+        /**
+        * Deletes created applications
+        */
+                afterAll(async () => {
```
Comment (flacatus) [inline tests/gpts/github/test-config/github_advanced_scenario.ts:415]: indent

Comment (jkopriva) [reply tests/gpts/github/test-config/github_advanced_scenario.ts]: Fixed

File: src/apis/kubernetes/kube.ts (hunk 2 above)
Comment (jkopriva) [reply src/apis/kubernetes/kube.ts]: Fixed

**Pull request merged**